### ISSUE TITLE
PhoneGap Build Error (Android)

### ISSUE THREAD

**Issue by daksamedia** [EddyVerbruggen/cordova-plugin-googleplus#85]
Hi Eddy,
Everything was fine before I added your plugin to my config.xml

```html
<gap:plugin name="cordova-plugin-googleplus" source="npm"/>
```
After all I added this code above, everything was in trouble (on Android only)

![pb-error](https://cloud.githubusercontent.com/assets/8596590/8894875/45156ce2-33f0-11e5-97fd-2826db7762b0.PNG)

Hopefully you can fix this issue.

Thanks.

**Comment by josschne** [comment]
@daksamedia You'll have to provide more details on your error before anyone will be able to help you out.

**Comment by daksamedia** [comment]
@josschne , I was clearly explaining my issue.
When I put this line 
```html
<gap:plugin name="cordova-plugin-googleplus" source="npm"/>
```
I couldn't be able to get my app build on Android.
(As you can see on my attached screen above)

I'm using PhoneGap Build to develop my app.
Could you tell me what the really happen there??

**Comment by raphaelduartepinheiro** [comment]
@daksamedia, please be polite when asking for help. And no, you were unclear explaining the situation. Have you clicked on "Error" to see more details and provide it on this issue?

**Comment by daksamedia** [comment]
I'm so sorry, it was my bad.
And this is what the phonegap builder site said to me :
```HTML
"Oh geez. Your build failed. Sorry, but a problem occurred on the build server"
```

Thank so much

**Comment by josschne** [comment]
@daksamedia Can you please click on "Log" and provide the build log for the failed build?

**Comment by mynk28** [comment]
I have Same Issue any Solution...
Here is my error Log 
https://build.phonegap.com/apps/1476759/logs/android/build/

**Comment by wongjoh** [comment]
Same issue (pretty much identical log file as well). Any ideas?

**Comment by josschne** [comment]
@wongjoh @mynk28 Can you please post the log from a local (non PGBuild) build? I'd like to determine if the behavior is different when you build locally.

**Comment by wongjoh** [comment]
When I build locally there are no issues and am able to obtain a local APK, but then again it is a different process (using 'phonegap build android --release' and then jarsigner). I have not done the zipalign thing since it was just for development testing

**Comment by josschne** [comment]
@wongjoh Thanks for that datapoint. I am wondering if the issue has to do with the PhoneGap build environment up at Adobe. I've only used the service briefly, so I'm not sure how much of the Android SDK is installed when they setup their environment. It's very possible they simply don't have the right libraries from the SDK installed in that environment.


I'd suggest contacting Adobe and ask the following:

Are the Android SDK packages "Extras/Android Support Repository" and "Extras/Google Repository" installed in your build environment? If not, is there a way to enable them?

**Comment by raphaelduartepinheiro** [comment]
Maybe a helpful tip:
I was not able to install ``cordova-plugin-googleplus`` with older versions of Cordova which use Ant as build platform. I've needed to migrate to Cordova 5.0 and build with Gradle.

**Comment by mohitmayank** [comment]
folks. I am getting the same issue. I am using cordova 5.1.1. and building locally on a ubuntu machine. 

```
[mohit@M cordova]$ cordova build android
Running command: /home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/build 
/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:126
                    throw e;
                          ^
Error: ENOENT, no such file or directory '/home/mohit/bin/android-studio/sdk/com.google.android.gms:play-services-plus/build.xml'
    at Error (native)
    at Object.fs.openSync (fs.js:500:18)
    at Object.fs.writeFileSync (fs.js:1099:15)
    at writeBuildXml (/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/lib/build.js:96:24)
    at /home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/lib/build.js:104:21
    at _fulfilled (/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:798:54)
    at self.promiseDispatch.done (/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:827:30)
    at Promise.promise.promiseDispatch (/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:760:13)
    at /home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:574:44
    at flush (/home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/node_modules/q/q.js:108:17)
ERROR building one of the platforms: Error: /home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/build: Command failed with exit code 1
You may not have the required environment or OS to build this project
Error: /home/mohit/Documents/findYogi/git/fyapp/cordova/platforms/android/cordova/build: Command failed with exit code 1
    at ChildProcess.whenDone (/usr/local/lib/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:134:23)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1015:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
```

**Comment by wongjoh** [comment]
Famous official response from PG Moderator:
" If you believe it's a missing piece need to be added to the builder, then please file feature request below: https://github.com/phonegap/build/issues
Otherwise you might need to contact plugin's owner on the actual root cause or available workaround."

My report (which was deemed as 'Not a Problem' ;-) is here: http://community.phonegap.com/nitobi/topics/android-building-error-with-plugin-google-plus-com-google-android-gms-play-services-plus

PG Build log is here: https://build.phonegap.com/apps/1542795/logs/android/build/

Are others able to do Phonegap Build when using the latest versions of the plugin? I'm relatively new to this so not sure if I'm the one with the wrong setup/etc

**Comment by raphaelduartepinheiro** [comment]
@mohitmayank it has occurred with me. `cordova platform rm android` and add the platform again, it will now come with the Gradle build system wichi suit better with Cordova > 5.

**Comment by mohitmayank** [comment]
@raphaelduartepinheiro Yes. Thanks for the tip.  i figured that too. And finally got everything working with cordova 5.0.0

**Comment by EddyVerbruggen** [comment]
Hey guys, just released the 4.0.0 version to npm. This release should work with any Cordova version supported on build.phonegap.com. Add it to your project this way:

```xml
<gap:plugin name="cordova-plugin-googleplus" source="npm">
  <param name="REVERSED_CLIENT_ID" value="myreversedclientid" />
</gap:plugin>
```

You can read about how to obtain the `REVERSED_CLIENT_ID` in the [readme](https://github.com/EddyVerbruggen). It makes the `iosApiKey` obsolete.
